### ISSUE TITLE
fs.pathExistsSync is not a function

### ISSUE THREAD

**Issue by lastmjs** [graphcool/graphcool#875]
When I run `graphcool deploy` I get the following error in the terminal:

```
fs.pathExistsSync is not a function
```

My graphcool.yml has the following format for functions:

```
functions:
    assignment-lti-launch:
        handler:
            code:
                src: ./functions/assignment-lti-launch/assignment-lti-launch.ts
                environment:
                    PRENDUS_LTI_SECRET: ${env:PRENDUS_LTI_SECRET}
        type: resolver
        schema: ./functions/assignment-lti-launch/assignment-lti-launch.graphql
```

When I deploy with the functions defined as above, I get the error. When I remove the functions section, everything deploys fine

**Comment by marktani** [comment]
Are you deploying locally or to a shared cluster?

**Comment by lastmjs** [comment]
I'm deploying to my project on graph.cool

**Comment by lastmjs** [comment]
Hold on, maybe I'm confused. If I do an `npm install graphcool@next` right now, am I supposed to get version 0.7.0?

**Comment by marktani** [comment]
Yes, you are.

**Comment by lastmjs** [comment]
Alright, it looks like my graphcool.yml follows the schema, and I'm on version 0.7.0, and yes this error continues to appear.

**Comment by lastmjs** [comment]
Also, the error continues to appear even if the function is a `.js` file instead of a `.ts` file.

**Comment by lastmjs** [comment]
The issue goes away if I install `graphcool@0.7.0-beta.12`

**Comment by kbrandwijk** [comment]
Okay, very complicated scenario, but @lastmjs and me figured it out. First of all, the `fs-extra` dependency is missing from the `graphcool-cli-core` package. This is the main cause of this problem.

Now what happens? Once `graphcool-cli-core` requires `fs-extra`, it starts looking for it. It will not find it in its own `node_modules` folder, and will look one folder up, where it was installed, because it is required by `graphcool-cli` and `graphcool-cli-engine`. So everything works... :tada:

...**Except**, if you happen to have another project dependency that uses a different version of `fs-extra`, which was the case. Then, that version is already installed in `node_modules`, and when you install `graphcool`, `npm` does some conflict resolution and places the conflicting dependency for `graphcool` in it's `node_modules` folder. So it ends up in `graphcool-cli/node_modules` and `graphcool-cli-engine/node_modules`, but NOT in `graphcool-cli-core/node_modules`, because that package doesn't have the dependency specified. At runtime, the resolved version for all `require` statements in `graphcool-cli-core` will be the one from the top `node_modules` folder.

This happened to be `fs-extra 0.30.0`, which does not include `fs.pathExistsSync()`.

So the bottom line is: _Please check the package.json dependencies for all CLI components_

**Comment by timsuchanek** [comment]
Thanks for researching this already so well @kbrandwijk @lastmjs @marktani ! I will add `fs-extra` to the cli core and check the others dependencies.

**Comment by timsuchanek** [comment]
This is fixed and published at `graphcool@0.8.0-alpha.8`
